### DISCUSSION TITLE
request() takes 1 positional argument but 2 were given

### DISCUSSION THREAD

**Discussion by deardaming** [mitmproxy/mitmproxy#5452] [Q&A]
#### Problem Description
When I start using the command mitmdump -s main.py then this error occurs

-------------------------------------------------- [error]
Addon error: Traceback (most recent call last):
  File "d:\python388\lib\site-packages\mitmproxy\addonmanager.py", line 43, in safecall
    yield
  File "d:\python388\lib\site-packages\mitmproxy\addonmanager.py", line 288, in trigger_event
    await self.invoke_addon(i, event)
  File "d:\python388\lib\site-packages\mitmproxy\addonmanager.py", line 265, in invoke_addon
    res = func(*event.args())
TypeError: request() takes 1 positional argument but 2 were given

#### System Information
Paste the output of "mitmproxy --version" here. [1]

Mitmproxy: 8.0.0
Python:    3.8.8
OpenSSL:   OpenSSL 1.1.1n  15 Mar 2022
Platform:  Windows-10-10.0.22000-SP0

-------------------------------------------------- [2]

--> mitmdump --version

Mitmproxy: 8.0.0
Python:    3.8.8
OpenSSL:   OpenSSL 1.1.1n  15 Mar 2022
Platform:  Windows-10-10.0.22000-SP0
-------------------------------------------------- [3]

--> python -V
Python 3.8.8

-------------------------------------------------- [4]

--> my main.py

class Counter:
    def __init__(self):
        self.num = 0

    # def request(self, flow):
    #     self.num = self.num + 1
    #     ctx.log.info("We've seen %d flows" % self.num)
    

    def request(flow: http.HTTPFlow):
        print('--> ',flow.request.pretty_url)
        if 'http' in flow.request.pretty_url:
            http.Response.make(
                status_code=200,
                content=b'123',
                headers={}
            )

addons = [Counter()]

**Comment by deardaming** [comment]
help pls , is it problem with my python version?

**Comment by Prinzhorn** [comment]
`self`
